tests/components/fibaro/test_config_flow.py

```diff
@@ -222,3 +224,124 @@ async def test_config_flow_import(hass):
             CONF_PASSWORD: TEST_PASSWORD,
             CONF_IMPORT_PLUGINS: False,
         }
+
+
+async def test_reauth_success(hass):
+    """Successful reauth flow initialized by the user."""
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id=TEST_SERIALNUMBER,
+        data={
+            CONF_URL: TEST_URL,
+            CONF_USERNAME: TEST_USERNAME,
+            CONF_PASSWORD: TEST_PASSWORD,
+            CONF_IMPORT_PLUGINS: False,
+        },
+    )
+    mock_config.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": mock_config.entry_id,
+        },
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "reauth_confirm"
+    assert result["errors"] == {}
+
+    login_mock = Mock()
+    login_mock.get.return_value = Mock(status=True)
+    with patch(
+        "fiblary3.client.v4.client.Client.login", login_mock, create=True
```

Comment:
Usually we patch the library where it is being tested instead of globally in tests to keep the patch as targeted as possible

Reply:
Unfortunately I have no idea what you mean with your comment. Can you describe your idea with some more details?

Basically this mocks the entry point of the library which I thought is the correct way. But I'm a beginner in writing python mock tests so I'm happy to learn more.

Reply:
```diff
diff --git a/tests/components/fibaro/test_config_flow.py b/tests/components/fibaro/test_config_flow.py
index ce00bf0d6e..327dd08f76 100644
--- a/tests/components/fibaro/test_config_flow.py
+++ b/tests/components/fibaro/test_config_flow.py
@@ -33,20 +33,20 @@ def fibaro_client_fixture():
     client_mock = Mock()
     client_mock.base_url.return_value = TEST_URL
 
-    with patch("fiblary3.client.v4.client.Client.__init__", return_value=None,), patch(
-        "fiblary3.client.v4.client.Client.info",
+    with patch("homeassistant.components.fibaro.FibaroClientV4.__init__", return_value=None,), patch(
+        "homeassistant.components.fibaro.FibaroClientV4.info",
         info_mock,
         create=True,
-    ), patch("fiblary3.client.v4.client.Client.rooms", array_mock, create=True,), patch(
-        "fiblary3.client.v4.client.Client.devices",
+    ), patch("homeassistant.components.fibaro.FibaroClientV4.rooms", array_mock, create=True,), patch(
+        "homeassistant.components.fibaro.FibaroClientV4.devices",
         array_mock,
         create=True,
     ), patch(
-        "fiblary3.client.v4.client.Client.scenes",
+        "homeassistant.components.fibaro.FibaroClientV4.scenes",
         array_mock,
         create=True,
     ), patch(
-        "fiblary3.client.v4.client.Client.client",
+        "homeassistant.components.fibaro.FibaroClientV4.client",
         client_mock,
         create=True,
     ):
@@ -66,7 +66,7 @@ async def test_config_flow_user_initiated_success(hass):
     login_mock = Mock()
     login_mock.get.return_value = Mock(status=True)
     with patch(
-        "fiblary3.client.v4.client.Client.login", login_mock, create=True
+        "homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True
     ), patch(
         "homeassistant.components.fibaro.async_setup_entry",
         return_value=True,
@@ -102,7 +102,7 @@ async def test_config_flow_user_initiated_connect_failure(hass):
 
     login_mock = Mock()
     login_mock.get.return_value = Mock(status=False)
-    with patch("fiblary3.client.v4.client.Client.login", login_mock, create=True):
+    with patch("homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True):
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
             {
@@ -129,7 +129,7 @@ async def test_config_flow_user_initiated_auth_failure(hass):
 
     login_mock = Mock()
     login_mock.get.side_effect = HTTPException(details="Forbidden")
-    with patch("fiblary3.client.v4.client.Client.login", login_mock, create=True):
+    with patch("homeassistant.components.fibaro.FibaroClientV4login", login_mock, create=True):
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
             {
@@ -156,7 +156,7 @@ async def test_config_flow_user_initiated_unknown_failure_1(hass):
 
     login_mock = Mock()
     login_mock.get.side_effect = HTTPException(details="Any")
-    with patch("fiblary3.client.v4.client.Client.login", login_mock, create=True):
+    with patch("homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True):
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
             {
@@ -200,7 +200,7 @@ async def test_config_flow_import(hass):
     login_mock = Mock()
     login_mock.get.return_value = Mock(status=True)
     with patch(
-        "fiblary3.client.v4.client.Client.login", login_mock, create=True
+        "homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True
     ), patch(
         "homeassistant.components.fibaro.async_setup_entry",
         return_value=True,
@@ -255,7 +255,7 @@ async def test_reauth_success(hass):
     login_mock = Mock()
     login_mock.get.return_value = Mock(status=True)
     with patch(
-        "fiblary3.client.v4.client.Client.login", login_mock, create=True
+        "homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True
     ), patch(
         "homeassistant.components.fibaro.async_setup_entry",
         return_value=True,
@@ -297,7 +297,7 @@ async def test_reauth_connect_failure(hass):
 
     login_mock = Mock()
     login_mock.get.return_value = Mock(status=False)
-    with patch("fiblary3.client.v4.client.Client.login", login_mock, create=True):
+    with patch("homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True):
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
             user_input={CONF_PASSWORD: "other_fake_password"},
@@ -336,7 +336,7 @@ async def test_reauth_auth_failure(hass):
 
     login_mock = Mock()
     login_mock.get.side_effect = HTTPException(details="Forbidden")
-    with patch("fiblary3.client.v4.client.Client.login", login_mock, create=True):
+    with patch("homeassistant.components.fibaro.FibaroClientV4.login", login_mock, create=True):
         result = await hass.config_entries.flow.async_configure(
             result["flow_id"],
             user_input={CONF_PASSWORD: "other_fake_password"},
```

Reply:
Ok understood, this makes sense. I changed it.